### PR TITLE
LOOFAH-53 introduce a DataRetriever interface to abstract mongodb from service layer

### PR DESCRIPTION
**Pull request by elishkaflint**

### CODE REVIEW

File: src/test/java/com/loofah/graph/api/retrievers/MongoBackedGradeRepositoryTest.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class MongoBackedGradeRepositoryTest {
```
Comment (Learner5200) [inline src/test/java/com/loofah/graph/api/retrievers/MongoBackedGradeRepositoryTest.java]: Nitpick: should this be `MongoBackedDataRetrieverTest`?

File: src/test/java/com/loofah/graph/api/retrievers/MongoBackedGradeRepositoryTest.java [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    private GradeRepository gradeRepository;
+
+    @Mock
+    private CraftRepository craftRepository;
+
+    @InjectMocks
+    private MongoBackedDataRetriever mongoBackedDataRetriever;
+
+    @Test
+    public void getSkillById_getsSkillViaDataRetriever() {
```
Comment (Learner5200) [inline src/test/java/com/loofah/graph/api/retrievers/MongoBackedGradeRepositoryTest.java]: Nitpick: should these be `_getsXViaRepository`? (or I guess `getsSkillsViaMongoQuery` or something for the test below this one)

**Review by Learner5200 — APPROVED** [review]

File: src/test/java/com/loofah/graph/api/retrievers/MongoBackedGradeRepositoryTest.java (hunk 2 above)
Comment (elishkaflint) [reply src/test/java/com/loofah/graph/api/retrievers/MongoBackedGradeRepositoryTest.java]: yep - updated :)

File: src/test/java/com/loofah/graph/api/retrievers/MongoBackedGradeRepositoryTest.java (hunk 1 above)
Comment (elishkaflint) [reply src/test/java/com/loofah/graph/api/retrievers/MongoBackedGradeRepositoryTest.java]: yep good spot - updated

**Pull request merged**